tv-show/src/components/features/shows/ShowReviewSection/ShowReviewSection.tsx

```diff
@@ -0,0 +1,82 @@
+import { IReview } from "@/typings/Review.type";
+import { Box, Card, CardBody, Container, Heading, Stack } from "@chakra-ui/react";
+import ReviewList from "../../review/ReviewList/ReviewList";
+import ReviewForm from "../../review/ReviewForm/ReviewForm";
+import { useEffect, useState } from "react";
+
+const LOCAL_STORAGE_KEY = 'reviews';
+
+function loadReviewsFromLocalStorage(): IReview[] {
+  const reviewsString = localStorage.getItem(LOCAL_STORAGE_KEY);
+  if(!reviewsString) {
+    return [];
+  }
+  return JSON.parse(reviewsString);
+}
+
+function storeReviewsToLocalStorage(reviews: IReview[]) {
+  localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(reviews));
+}
+
+function calculateAverageRating(reviews: IReview[]) {
+  let averageRating = 0;
+
+  reviews.forEach((review) => averageRating += review.rating);
+  if(reviews.length) {
+    averageRating /= reviews.length;
+  }
+
+  return averageRating ? averageRating : undefined;
+}
+
+interface IShowReviewSection {
+  setAverageRating: (avg: number | undefined) => void;
```

Comment:
Do you really want to pass undefined here?

Reply:
I wanted to differentiate between 0 and `undefined`, although in this case they would act the same

Reply:
If you have a use case where you really want to call `setAverageRating(undefined)` then this typing is OK. If not, you are just opening an avenue for errors.